src/domain/users/ip-metadata-validator.ts

```diff
@@ -0,0 +1,33 @@
+import { MissingIpMetadataError, InvalidIpMetadataTypeError } from "@domain/errors"
+import { getBlacklistedASNs, getWhitelistedCountries } from "@config"
+
+export const IpMetadataValidator = (): IpMetadataValidator => {
+  const validate = (userIP: UserIPs): true | ApplicationError => {
```

Comment:
we often use `validate` in the context of syntax validation. I'm wondering if something like `isApproved` or `approve` would not be more appropriate?

Reply:
I do agree here, I think `approve` would be a superior method name to `validate`. In this instance, I was mirroring the naming convention in `PhoneMetadataValidator().validate()`.

Would you like mot change both the `PhoneMetadataValidator()` and `IpMetadataValidator()` to have a method name of `approve`?

Reply:
Hmm so we are considering differentiating validation based on value construction vs validation based on value association.

I don't think we have made this differentiation so far - I don't mind but if change the verb `validate` we should also change the noun `Validator`.
As `approve` is a bit opaque (just like `validate`) perhaps a more explicit name like `isIpAllowedToLogIn` would also fix the ambiguity.